meta-balena-common/recipes-containers/balena-supervisor/balena-supervisor/update-balena-supervisor

```diff
@@ -99,30 +116,39 @@ fi
 echo "Getting image name and version..."
 if [ -n "$API_ENDPOINT" ] && [ -n "${UUID}" ] && [ -n "$_device_api_key" ]; then
     _target_state=$(os_helpers_fetch_target_state "${UUID}" "${API_ENDPOINT}" "${_device_api_key}")
-    read -r version image_name <<< "${_target_state}"
-    if [ -z "$version" ] || [ -z "$image_name" ]; then
-        echo "ERROR: Unexpected API data."
-        error_handler "unexpected API data"
+    if [ -n "${_target_state}" ]; then
+        read -r version image_name <<< "${_target_state}"
+        if [ -z "$version" ] || [ -z "$image_name" ]; then
+            echo "ERROR: Unexpected API data."
+            error_handler "unexpected API data"
+        fi
     fi
 fi
 
 # shellcheck disable=SC1091
 . /etc/balena-supervisor/supervisor.conf
 
 # If no API version is set, use preloaded values
-if [ "$version" = "null" ] || [ "$image_name" = "null" ]; then
-    echo "No supervisor configuration found from API. Using preloaded values."
-    if [ -z "$SUPERVISOR_VERSION" ]; then
-        echo "ERROR: No preloaded version found."
-        error_handler "no preloaded version found"
-    fi
-    if [ -z "$SUPERVISOR_IMAGE" ]; then
-        echo "ERROR: No preloaded image found."
-        error_handler "no preloaded image found."
+if [ -z "$version" ] || [ -z "$image_name" ] || [ "$version" = "null" ] || [ "$image_name" = "null" ]; then
+    echo "No supervisor configuration found from API."
+    if [ -n "${TARGET_SUPERVISOR_IMAGE}" ]; then
+        echo "Using command line image argument $TARGET_SUPERVISOR_IMAGE."
+        image_name="${TARGET_SUPERVISOR_IMAGE%:*}"
```

Comment:
We don't need to strip the colon here if we validate the input above.

Reply:
Agreed